packages/flutter/lib/src/material/outline_button.dart

```diff
@@ -355,21 +355,21 @@ class _OutlineButtonState extends State<OutlineButton> with SingleTickerProvider
   }
 
   BorderSide _getOutline(ThemeData theme, ButtonThemeData buttonTheme) {
-    final bool themeIsDark = theme.brightness == Brightness.dark;
     if (widget.borderSide?.style == BorderStyle.none)
       return widget.borderSide;
 
+    // TODO(antrob): Replace primaryColor with surfaceColor.
+    final Color themeColor = theme.primaryColor ??
+        theme.brightness == Brightness.dark ? Colors.white : Colors.black;
     final Color color = widget.enabled
       ? (_pressed
-         ? widget.highlightedBorderColor ?? theme.primaryColor
-         : (widget.borderSide?.color ??
-            (themeIsDark ? Colors.grey[600] : Colors.grey[200])))
-      : (widget.disabledBorderColor ??
-         (themeIsDark ? Colors.grey[800] : Colors.grey[100]));
+          ? widget.highlightedBorderColor ?? themeColor
+          : (widget.borderSide?.color ?? themeColor))
+      : (widget.disabledBorderColor ?? themeColor);
```

Comment:
i'm even more confused by this code now :-(

Reply:
OK. HansMuller pointed out a good unknown about disabled state, so I am waiting to hear back from the designers.

Reply:
The general style pattern for a stateless widget appears to be:

1. Use any provided overrides coming from the ctor.
2. If there are no overrides, use the appropriate custom theme value.
3. If there is no custom theme, use the baseline material theme value.

There may also be state specific overrides, (pressed, disabled, etc). It should follow the same pattern, but I am clarifying these defaults since 'disabled' state is a bit ambiguous in the spec.

Reply:
@HansMuller

Reply:
That's right about the general style here.